packages/dataprovider/src/buildVariables.test.ts

```diff
@@ -176,6 +228,82 @@ describe("buildVariables", () => {
         },
       });
     });
+
+    it("correctly connects new socialmedia", () => {
+      const params = {
+        data: {
+          id: "einstein",
+          userSocialMedia: { id: "newId", twitter: "tw", instagram: "in" },
+        },
+        previousData: {
+          userSocialMedia: { id: "oldId", twitter: "t", instagram: "i" },
+        },
+      };
+
+      expect(
+        buildVariables(testIntrospection)(testUserResource, UPDATE, params),
+      ).toEqual<NexusGenArgTypes["Mutation"]["updateOneUser"]>({
+        where: { id: "einstein" },
+        data: {
+          userSocialMedia: {
+            connect: { id: "newId" },
+          },
+        },
+      });
+    });
+
+    it("correctly update socialmedia", () => {
+      const params = {
+        data: {
+          id: "einstein",
+          userSocialMedia: {
+            id: "socialId",
+            twitter: "new-twitter",
+            instagram: "another-instagram",
+          },
+        },
+        previousData: {
+          userSocialMedia: {
+            id: "socialId",
+            twitter: "twitter",
+            instagram: "instagram",
+          },
+        },
+      };
+
+      expect(
+        buildVariables(testIntrospection)(testUserResource, UPDATE, params),
+      ).toEqual<NexusGenArgTypes["Mutation"]["updateOneUser"]>({
+        where: { id: "einstein" },
+        data: {
+          userSocialMedia: {
+            update: { twitter: "new-twitter", instagram: "another-instagram" },
+          },
+        },
+      });
+    });
+
+    it("correctly disconnects socialmedia", () => {
+      const params = {
+        data: {
+          id: "einstein",
+        },
+        previousData: {
+          userSocialMedia: { id: "oldId", twitter: "t", instagram: "i" },
+        },
+      };
+
+      expect(
+        buildVariables(testIntrospection)(testUserResource, UPDATE, params),
+      ).toEqual<NexusGenArgTypes["Mutation"]["updateOneUser"]>({
+        where: { id: "einstein" },
+        data: {
+          userSocialMedia: {
+            disconnect: true,
+          },
+        },
+      });
+    });
```

Comment:
we should also add:

update with update many relations
update with create new many relations